src/pytorch_lightning/plugins/io/wrapper.py

```diff
@@ -0,0 +1,45 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+
+
+class _WrappingCheckpointIO:
```

Comment:
It might be better if `_WrappingCheckpointIO` was a subclass of `CheckpointIO`, so that the inheritance is easier. Plus, it can have the `{save,load,remove}_checkpoint` methods, which would automatically check for `self._base_checkpoint_io_configured == True`. And the implementations of this class (such as `AsyncCheckpointIO`) could call `super().save_checkpoint(...)` instead of `self._checkpoint_io.save_checkpoint(...)` which is arguably more pythonic since it doesn't use private attributes

Reply:
I can still access it using `self.checkpoint_io.load_checkpoint()` since there's a `checkpoint_io` getter in `_WrappingCheckpointIO`. WDYT? I am hesitating to add `CheckpointIO` as the base because it's not actually a `CheckpointIO` class.

Reply:
I think the main benefit from my point of view would be automatic check, whether `self._base_checkpoint_io_configured == True` and simplified inheritance (the implementations would not need to inherit from two classes but just one.)

Reply:
I agree with Ota

Reply:
done!